crates/network-scanner/examples/block_boardcast.rs

```diff
@@ -0,0 +1,36 @@
+use std::{process::exit, time::Duration};
+
+use network_scanner::boardcast::block_broadcast;
+
+pub fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::SubscriberBuilder::default()
+        .with_max_level(tracing::Level::TRACE)
+        .init();
+
+    let ip = std::net::Ipv4Addr::new(192, 168, 1, 255);
+
+    let iterator = block_broadcast(ip, Some(Duration::from_secs(1)))?;
+
+    for result in iterator {
+        match result {
+            Ok(ping_res) => {
+                println!("received result {:?}", &ping_res);
+            }
+            Err(e) => match e.downcast_ref::<std::io::Error>() {
+                Some(e) => match e.kind() {
+                    std::io::ErrorKind::TimedOut => {
+                        tracing::info!("Timed out");
+                        exit(0);
```

Comment:
issue: `exit(0)` is not running any destructor and is generally not necessary.

> Note that because this function never returns, and that it terminates the process, no destructors on the current stack or any other thread’s stack will be run. If a clean shutdown is needed it is recommended to only call this function at a known point where there are no more destructors left to run; or, preferably, simply return a type implementing [Termination](https://doc.rust-lang.org/std/process/trait.Termination.html) (such as [ExitCode](https://doc.rust-lang.org/std/process/struct.ExitCode.html) or Result) from the main function and avoid this function altogether

suggestion: Use `return Ok(())` if you want to exit with a success status code. Alternatively, you used `break` at the other places, which have the same behavior. Yet another alternative is to just return the error instead of handling it this way. I don’t really see any special logic happening other than breaking out of the loop and return `Ok(())` anyway. Maybe you didn’t mean to break the loop at the other places? (In order to continue processing of the remaining results.)

As an aside, a timeout can probably be considered an error too. (`Err` instead of `Ok`.)

Reply:
updated.

As of the timeout error, it is important to notice that timeout is different from other error. When searching the network, we do not know exactly how long it takes for some computer to respond back to us, so it is necessary to set a timeout for the operation. if the error kind is timeout, that means this is operation is finished as expected (there is no other way to finish it).